include/rocksdb/table.h

```diff
@@ -287,47 +306,88 @@ struct BlockBasedTableOptions {
-  // the memory, if block cache available.
+  // `cache_usage_options` allows users to specify the default
+  // options (`cache_usage_options.default_options`) and the overrides
+  // options (`cache_usage_options.options_overrides`)
+  // for different `CacheEntryRole` under various features related to cache
+  // usage.
+  //
+  // For a certain `CacheEntryRole r` and a certain feature `f` of
+  // `CacheEntryRoleOptions`:
+  // (a) If `options_overrides[static_cast<uint32_t>(role)].f !=
+  // Decision::kFallback`, then it will be used to override the `Decision` value
+  // of `cache_usage_options.default_options.f`.
+  //
+  // Example:
+  // (a.1)
+  // `options_overrides[static_cast<uint32_t>(CacheEntryRole::kBlockBasedTableReader)]
+  // = {.charged = Decision::kEnabled}`
+  // (a.2) `cache_usage_options.default_options.charged = Decision::kFallback`
+  // (a.1) + (a.2) => the `CacheEntryRoleOptions::charged`
+  // feature will be enabled for `CacheEntryRole::kBlockBasedTableReader`
+  //
+  // (b) If `options_overrides[static_cast<uint32_t>(role)].f ==
+  // Decision::kFallback`, then the `Decision` value of
+  // `cache_usage_options.default_options.f` will be used. If this default value
+  // is set to `Decision::kFallback`, then the existing behavior will be in
+  // effect. See "Features" below for the existing behaviors.
+  //
+  // Example:
+  // (b.1)
+  // `options_overrides[static_cast<uint32_t>(CacheEntryRole::kBlockBasedTableReader)]
+  // = {.charged = Decision::kFallback}`
+  // (b.2) `cache_usage_options.default_options.charged = Decision::kFallback`
+  // (b.1) + (b.2) => the `CacheEntryRoleOptions::charged` feature will fallback
+  // to the existing behavior for `CacheEntryRole::kBlockBasedTableReader`
+  //
+  // Features:
+  //
+  // `cache_usage_options` currently supports specifying options for the
+  // following features:
+  //
+  // 1. Memory charging to block cache (`CacheEntryRoleOptions::charged`)
+  //
+  // Memory charging is a feature of accounting memory usage of specific area
+  // (represented by `CacheEntryRole`) toward usage in block cache (if
+  // available), by updating a dynamical charge to the block cache loosely based
+  // on the actual memory usage of that area.
+  //
+  // Existing behavior:
+  //
+  // (1) CacheEntryRole::kCompressionDictionaryBuildingBuffer : kEnabled
+  // Charge memory usage of the buffered data used as training samples for
+  // dictionary compression.
+  // If such memory usage exceeds the avaible space left in the block cache
+  // at some point (i.e, causing a cache full under
+  // `LRUCacheOptions::strict_capacity_limit` = true), the data will then be
+  // unbuffered.
   //
-  // Charged memory usage includes:
-  // 1. Bloom Filter (format_version >= 5) and Ribbon Filter construction
-  // 2. More to come...
+  // (2) CacheEntryRole::kFilterConstruction : kDisabled
+  // If enabled, charge memory usage of Bloom Filter (format_version >= 5) and
```

Comment:
Since this is describing existing behavior I'd say something like "Same as kDisabled - does not charge memory usage of ...".

Reply:
Will fix!

Reply:
Fixed